lxc/config/remote.go

```diff
@@ -78,12 +78,20 @@ func (c *Config) GetInstanceServer(name string) (lxd.InstanceServer, error) {
 		d, err := lxd.ConnectLXDUnix(strings.TrimPrefix(strings.TrimPrefix(remote.Addr, "unix:"), "//"), args)
 		if err != nil {
 			var netErr *net.OpError
+			socketPath := ""
 
 			if errors.As(err, &netErr) {
-				return nil, fmt.Errorf("The LXD daemon doesn't appear to be started (socket path: %s)", netErr.Addr)
+				socketPath = fmt.Sprintf(" (socket path: %s)", netErr.Addr)
 			}
 
-			return nil, err
+			switch {
+			case errors.Is(err, os.ErrNotExist):
+				return nil, fmt.Errorf("The LXD daemon doesn't appear to be started%s", socketPath)
+			case errors.Is(err, os.ErrPermission):
+				return nil, fmt.Errorf("Permission denied%s", socketPath)
+			default:
+				return nil, err
```

Comment:
Also here lets add error wrapping as follows:

```go
return nil, fmt.Errorf("LXD unix socket file %q not accessible: %w", netErr.Addr, err)
```

Reply:
While having the error be `Unwrap`able by the caller would be ideal, the message gets pretty gnarly:
```
Error: LXD unix socket "/var/lib/lxd/unix.socket" not accessible: Get "http://unix.socket/1.0": dial unix /var/lib/lxd/unix.socket: connect: permission denied
```
I'm happy to use `%w` if you'd like but I thought a more concise message is probably worth the slightly less useful `error`.

Reply:
I think we should only do the error wrapping for the unknown error scenario - i.e not a permission denied or not found errors which will have the nicer error messages. But we should prefix it with an explanation as to what the general problem is.